dart-impl/mpi/src/dart_team_group.c

```diff
@@ -705,16 +714,23 @@ dart_ret_t dart_myid(dart_global_unit_t *unitid)
 
 dart_ret_t dart_size(size_t *size)
 {
-  int s;
-  MPI_Comm_size (DART_COMM_WORLD, &s);
-  (*size) = s;
+  static int mpi_size = -1;
+  if (dart_initialized()) {
+    if (mpi_size == -1) {
+      MPI_Comm_size (DART_COMM_WORLD, &mpi_size);
+    }
+    (*size) = mpi_size;
+  } else {
+    *size = 0;
```

Comment:
Function always returns `OK`.
For example in this line, should return `DART_ERR_OTHER`